src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs

```diff
@@ -373,47 +313,29 @@ protected override void DetachCore()
                     break;
                 }
 
-                this.pendingTip = tip;
+                this.ExpectedTip = tip;
             }
 
-            if (pendingTipBefore != this.pendingTip)
-                this.logger.LogTrace("Pending tip changed to '{0}'.", this.pendingTip);
+            if (pendingTipBefore != this.ExpectedTip)
+                this.logger.LogTrace("Pending tip changed to '{0}'.", this.ExpectedTip);
 
-            if ((this.pendingTip != null) && !this.bestChainSelector.TrySetAvailableTip(this.AttachedPeer.Connection.Id, this.pendingTip))
+            if ((this.ExpectedTip != null) && !this.bestChainSelector.TrySetAvailableTip(this.AttachedPeer.Connection.Id, this.ExpectedTip))
                 this.InvalidHeaderReceived = true;
-            
-            ChainedHeader chainedPendingTip = this.pendingTip == null ? null : this.Chain.GetBlock(this.pendingTip.HashBlock);
+
+            ChainedHeader chainedPendingTip = this.ExpectedTip == null ? null : this.chain.GetBlock(this.ExpectedTip.HashBlock);
             if (chainedPendingTip != null)
             {
                 // This allows garbage collection to collect the duplicated pendingTip and ancestors.
-                this.pendingTip = chainedPendingTip;
+                this.ExpectedTip = chainedPendingTip;
             }
 
             // If we made any advancement or the sync is enforced by 'doTrySync'- continue syncing.
-            if (doTrySync || (this.pendingTip == null) || (pendingTipBefore == null) || (pendingTipBefore.HashBlock != this.pendingTip.HashBlock))
+            if (doTrySync || (this.ExpectedTip == null) || (pendingTipBefore == null) || (pendingTipBefore.HashBlock != this.ExpectedTip.HashBlock))
                 await this.TrySyncAsync().ConfigureAwait(false);
 
             this.logger.LogTrace("(-)");
         }
 
-        public void SetPendingTip(ChainedHeader newTip)
```

Comment:
Is this method not used?

Reply:
No. It used to be when BS behavior was using pending tip. Now we have 2 tips (one represents the last block that was sent to the peer and another represents last header received from the peer). 
BTC Core do that so we decided to copy that logic.